root/etc/e-smith/events/actions/restore-data-restic

```diff
@@ -78,16 +78,20 @@ if ($restic_repository eq '') {
 
 push(@opts, '--repo', $restic_repository);
 
-my @cmd = (CMD, @opts, 'restore', $time, '--target', $position);
+my $ret;
 
 if (@files) {  # restore selected files
+    my @cmd = (CMD, @opts, 'restore', $time, '--target', $position);
     foreach (@files) {
         push(@cmd, '--include', $_);
     } 
+    $ret = system(@cmd);
+} else {
+    # use tar to restore files with non-numeric uid/gid
+    my @cmd = (CMD, @opts, 'dump', $time, '/', "| tar -xvf - -C $position");
+    $ret = system(join(" ", @cmd));
```

Comment:
Are you sure `@cmd` arguments are protected against shell injection and correctly escaped?

Reply:
No, their are not.
For sure `$position` can be injected and will not correctly handle spaces.
But the parameter is not accessible from the the UI, since it's hard-coded inside the API: https://github.com/NethServer/nethserver-restore-data/blob/master/api/execute#L103